server/internal/circ/pool_test.go

```diff
@@ -8,7 +8,7 @@ import (
 
 func TestNewBytesPool(t *testing.T) {
 	bpool := NewBytesPool(256)
-	require.NotNil(t, bpool.pool)
+	require.NotNil(t, &bpool.pool)
```

Comment:
Good catch! Technically I think this will always return true anyway because `NewBytesPool` unconditionally returns a value. However, I have added the `*sync.Pool` fix for vetting and consistency.

Reply:
The above test failures appear to be historic, and likely the result of the way the tests have been written rather than the underlying code. I've opened it as issue #25 :)